app/validators/appointment_time_validator.rb

```diff
@@ -0,0 +1,15 @@
+class AppointmentTimeValidator < ActiveModel::Validator
+
+  def validate(record)
+    appointments = Appointment.where(user_id: record.user_id)
+    if record.start_time
+      appointments.each do |appointment|
+        if appointment != record
+          if appointment.start_time <= record.start_time && record.start_time <= appointment.end_time || record.start_time <= appointment.start_time && appointment.start_time <= record.end_time
+            record.errors.add(:start_time, "Wybrany termin jest już zajęty!")
```

Comment:
1. Use translations
2. It's not readable, if u can't simplify it, at least move conditions to separate methods